File: backend/src/main/java/com/bookbook/domain/rent/dto/RentResponseDto.java
// 25.08.03 현준
package com.bookbook.domain.rent.dto;

import com.bookbook.domain.rent.entity.RentStatus;
import lombok.Builder;

import java.time.LocalDateTime;

// 대여 글 단건 조회를 위한 DTO
@Builder
public record RentResponseDto(
        int id, // 글 ID
        Long lenderUserId, // 대여자 ID
        String title, // 글 제목
        String bookCondition, // 책 상태
        String bookImage, // 글쓴이가 올린 책 이미지 URL
        String address, // 사용자 주소
        String contents, // 대여 내용
        RentStatus rentStatus,

        String bookTitle, // 책 제목
        String author,
        String publisher,
        String category, // 책 카테고리
        String description, // 책 설명
        LocalDateTime createdDate,
        LocalDateTime modifiedDate,

        // 글쓴이 조회를 위해 필드 추가
        String nickname, // 글쓴이 닉네임
        Float rating, // 글쓴이 평점
        int lenderPostCount, // 글쓴이 대여글 작성 수
        
        // 찜 상태 확인을 위한 필드 추가
        boolean isWishlisted // 현재 사용자의 찜 상태

) {
}

File: backend/src/main/java/com/bookbook/domain/rent/dto/RentResponseDto.kt
// 25.08.03 현준
package com.bookbook.domain.rent.dto

import com.bookbook.domain.rent.entity.RentStatus
import java.time.LocalDateTime

// 25.08.28 현준
// 대여 게시글의 상세 정보를 담는 DTO.
data class RentResponseDto(
    val id: Int,                           // 글 ID
    val lenderUserId: Long,                // 대여자 ID
    val title: String,                     // 글 제목
    val bookCondition: String,             // 책 상태
    val bookImage: String,                 // 글쓴이가 올린 책 이미지 URL
    val address: String,                   // 사용자 주소
    val contents: String,                  // 대여 내용
    val rentStatus: RentStatus,            // 대여 상태

    // 책 정보
    val bookTitle: String,                 // 책 제목
    val author: String,                    // 책 저자
    val publisher: String,                 // 책 출판사
    val category: String,                  // 책 카테고리
    val description: String,               // 책 설명
    val createdDate: LocalDateTime?,       // 생성일
    val modifiedDate: LocalDateTime?,      // 수정일

    // 글쓴이 정보
    val nickname: String,                  // 글쓴이 닉네임
    val rating: Float?,                    // 글쓴이 평점
    val lenderPostCount: Int,              // 글쓴이 대여글 작성 수

    // 찜 상태 정보
    val isWishlisted: Boolean              // 현재 사용자의 찜 상태
)